### PR TITLE
updates for wave mesh cap in workflow

### LINKED ISSUE #736: Update wave related settings and code for P8c
**Description**
A new mesh based cap (work by @DeniseWorthen) will be used in p8c. We will be updating which grid the wave model is using.  Updates for the new grid and new code will be required. 
 
**Requirements**
This will require an update to ufs-weather-model (PR is still under review), will require a new input grid, and various setting updates.  There will likely also be build updates that are required. 

**Acceptance Criteria (Definition of Done)**
Done means we can run prototype 8c from the workflow with the new wave cap and grids, setting up the input files correctly (if we do not use multi, a shel input file will have to be created), setting updates will need to be made in addition to creation of regular lat/lon gridded output. 

**Dependencies**
https://github.com/ufs-community/ufs-weather-model/pull/1131

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
**Description**

Updates to work with the new WW3 mesh cap, including pointing to the new ufs-weather-model tag for p8c. 
The log files were removed for the non-pdgen wave script and ush files. 

Fixes #736 

**Type of change**

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [x] Clone and Build tests on WCOSS Dell P3
- [x] Coupled with aero forecast test on Orion
- [x] Coupled no-aero forecast  test on Hera

**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

File: modulefiles/module_base.hera.lua
Comment (WalterKolczynski-NOAA) [inline modulefiles/module_base.hera.lua]: Why was all this deleted? We still need all this to run verification.

File: parm/config/config.defaults.s2sw
Comment (WalterKolczynski-NOAA) [inline parm/config/config.defaults.s2sw]: Please remove the dead code

File: sorc/build_ufs.sh
Comment (WalterKolczynski-NOAA) [inline sorc/build_ufs.sh:11]: We can remove the 's' from getopts too.

File: sorc/checkout.sh
Comment (WalterKolczynski-NOAA) [inline sorc/checkout.sh]: This needs to be a hash or tag, not a branch

File: sorc/checkout.sh
Comment (WalterKolczynski-NOAA) [inline sorc/checkout.sh]: Is this going to be merged into UFS soon?

Also, this whole bit will need to be redone after the checkout refactor that just went in. Should be simple though.

File: ush/forecast_postdet.sh
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh:617]: Instead of setting a new variable beforehand, why not just determine multi-grid from grdALL?

File: ush/forecast_postdet.sh
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh:627]: Did WW3 change this to a more clumsy way of using a different suffix when there is only one grid? Or did we just always assume multi-grid before? Either way I'd love some consistency from WW3 here.

File: ush/forecast_postdet.sh
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh]: Can we do all these without using eval? I know that's how it was done before, but I'd like to reduce the number of evals in our scripts, not increase them.

File: ush/forecast_postdet.sh
Comment (WalterKolczynski-NOAA) [inline ush/forecast_postdet.sh]: I'd rather this be something more readable, like:
`if (( $( ls -1 $FIXwave/rmp_src_to_dst_conserv_* > /dev/null | wc -l) > 0 )); then`

It does the same thing, but it is clearer we are counting the number of files and checking against zero.

File: modulefiles/module_base.hera.lua
Comment (WalterKolczynski-NOAA) [inline modulefiles/module_base.hera.lua]: If we're going to update the compiler and module versions, we should do it for Orion too (Dell doesn't have the newer compiler and will be decommissioned soon).

File: sorc/build_ww3prepost.sh
Comment (WalterKolczynski-NOAA) [inline sorc/build_ww3prepost.sh]: When does -c get used? There is also no hook in build_all to turn it on. Does it need to be coordinated with build_ufs somehow?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
With the compiler and module changes, we'll need to test this uncoupled as well.

File: modulefiles/module_base.hera.lua
Comment (JessicaMeixner-NOAA) [reply modulefiles/module_base.hera.lua]: Actually the verification job ran? 

And there is no met/metplus for the updated intel and I thought I understood from you that this shouldn't hold up the PR? Maybe I'm mistaken?

File: sorc/checkout.sh
Comment (WalterKolczynski-NOAA) [reply sorc/checkout.sh]: Looks like you took care of this while I was reviewing.

File: sorc/checkout.sh
Comment (WalterKolczynski-NOAA) [reply sorc/checkout.sh]: Looks like you took care of this while I was reviewing.

Comment (JessicaMeixner-NOAA) [reply sorc/checkout.sh]: looks like the most recent code change changing this to a tag didn't get updated, it should be fixed with the merged in code now.

File: modulefiles/module_base.hera.lua
Comment (WalterKolczynski-NOAA) [reply modulefiles/module_base.hera.lua]: Looks like you took care of this while I was reviewing.

File: sorc/checkout.sh
Comment (JessicaMeixner-NOAA) [reply sorc/checkout.sh]: Yeah, somehow my last change reverting this didn't get in, it's changed to be pointing to the p8c tag now.

File: ush/forecast_postdet.sh
Comment (JessicaMeixner-NOAA) [reply ush/forecast_postdet.sh:627]: We always used multi-grid before.  Shel uses 'ww3' as the extension.  Even if WW3 changes, it's not going to be changed in time for this PR.

File: ush/forecast_postdet.sh
Comment (JessicaMeixner-NOAA) [reply ush/forecast_postdet.sh]: I will try to see if this works with out the eval and report back.  As you mentioned, this is how it was done before and not something new.

File: modulefiles/module_base.hera.lua
Comment (JessicaMeixner-NOAA) [reply modulefiles/module_base.hera.lua]: Orion has now been updated, an oversight.  Working on the testing there now.

File: ush/forecast_postdet.sh
Comment (JessicaMeixner-NOAA) [reply ush/forecast_postdet.sh:617]: It cannot be simply determined from grdALL and therefore requires a new variable.

File: modulefiles/module_base.hera.lua
Comment (WalterKolczynski-NOAA) [reply modulefiles/module_base.hera.lua]: I think verification "runs" but is just skipped for coupled. I think it may just be several cycles before it kicks in.

I definitely don't want to re-break verification right after I fixed it. I'm trying to push hpc-stack to get the PR in to add MET and MET+ to the official installations, but if you have to use the updated compiler and loading the old one at the end unloads the stuff you need, I can try to build MET and MET+ myself. Right now I'm pointing to a hacked version of the system one in a custom stack location.

File: ush/forecast_postdet.sh
Comment (WalterKolczynski-NOAA) [reply ush/forecast_postdet.sh:627]: I knew WW3 wouldn't change in time for this PR, just annoying we have to have all these ifs because the filename pattern changes. Could we instead have a variable like wave_suffix that equals wavGRD if multi-grid and 'ww3' otherwise? Then we can just always loop over that and eliminate the rest of these new ifs.

File: sorc/build_ww3prepost.sh
Comment (JessicaMeixner-NOAA) [reply sorc/build_ww3prepost.sh]: @WalterKolczynski-NOAA thanks for finding this... it was passed in before at one point, I guess it got messed up in a bad merge when the -c was removed from the build_all.  I'll have to re-think this now.

File: modulefiles/module_base.hera.lua
Comment (JessicaMeixner-NOAA) [reply modulefiles/module_base.hera.lua]: I don't want to break verification either, but it seems ran and loading the old intel is not going to work. Please let me know what the best path forward is.

File: sorc/build_ww3prepost.sh
Comment (WalterKolczynski-NOAA) [reply sorc/build_ww3prepost.sh]: Maybe use `-m` for 'meshcap' in `build_all.sh` instead, since `-c` is taken? Can easily turn -m for build_all into -c for build_ww3prepost.

Comment (WalterKolczynski-NOAA) [reply sorc/build_ww3prepost.sh]: Nevermind, the way you changed it to use APP is better.

Comment (JessicaMeixner-NOAA) [reply sorc/build_ww3prepost.sh]: It's based on the APP, so ATMW (pervious -c) versus S2S*, so I've updated this to mimic to ufs build options.

File: sorc/build_ww3prepost.sh
Comment (WalterKolczynski-NOAA) [inline sorc/build_ww3prepost.sh:23]: Does `$SWITCHFILE` down below need to be updated to use this variable?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: sorc/build_ww3prepost.sh
Comment (JessicaMeixner-NOAA) [reply sorc/build_ww3prepost.sh:23]: yes! thank you for catching this too!!

File: modulefiles/module_base.hera.lua
Comment (WalterKolczynski-NOAA) [reply modulefiles/module_base.hera.lua]: Okay, let me see what I can do about building MET and MET+.

**Comment by JessicaMeixner-NOAA** [comment]
@WalterKolczynski-NOAA tests with the updated code from review are ongoing.   I believe there is only one major unresolved issue which is the Met/MetPlus modules.  On orion there also was not the ufswm module for the hpc-miniconda3.  I'm not sure if that's also a Met/MetPlus verification requirement or something else?

**Comment by WalterKolczynski-NOAA** [comment]
> @WalterKolczynski-NOAA tests with the updated code from review are ongoing. I believe there is only one major unresolved issue which is the Met/MetPlus modules. On orion there also was not the ufswm module for the hpc-miniconda3. I'm not sure if that's also a Met/MetPlus verification requirement or something else?

That's why it was using the one I built. It's a requirement for aerosol init, and is what started the whole issue with verification in the first place.

**Comment by JessicaMeixner-NOAA** [comment]
Can we create a second set of modules for the aerosol init job and verification jobs to use the older version of intel with your hpc-stack modules?  Or has anyone started on the issue https://github.com/NOAA-EMC/global-workflow/issues/812 which is more/less what some of this is requiring/doing? 

The orion test is currently running (it was in the queue from yesterday until early this morning) 

And I ran both coupled with and without aerosol on hera: 
/scratch1/NCEPDEV/climate/Jessica.Meixner/p8c/reviews/wave01
/scratch1/NCEPDEV/climate/Jessica.Meixner/p8c/reviews/aero01

@WalterKolczynski-NOAA are there other tests I should be running? Should I be reaching out to the hpc-stack team that we have ufswm on hera but not on orion?  Metplus?

**Comment by JessicaMeixner-NOAA** [comment]
I have commented on the hpc-stack issue for the ufswm module on orion https://github.com/NOAA-EMC/hpc-stack/issues/449 
and for the PR for the MetPlus https://github.com/NOAA-EMC/hpc-stack/pull/324 about the outstanding module issues for this PR.

**Comment by JessicaMeixner-NOAA** [comment]
The experiment for rolling back the modules has started and it's output (com, etc) can be found here: 
/scratch1/NCEPDEV/climate/Jessica.Meixner/p8c/rollbackmods/roll01

**Comment by JessicaMeixner-NOAA** [comment]
The roll back test succeeded on hera, so I committed the changes to the branch here.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Approved pending testing.

**Comment by JessicaMeixner-NOAA** [comment]
The test on orion has finished: /work/noaa/marine/jmeixner/p8c/aero01 
the coupled_ic job failed because the wave ICs need to be resync'd after the permissions issues were fixed on Hera. I manually copied over the IC for the test.

**Pull request merged**
